### DISCUSSION TITLE
Pinecil v1 brick

### DISCUSSION THREAD

**Discussion by korzhan** [Ralim/IronOS#1815] [General]
I loaded the firmware into the soldering iron in bin format instead of dfu. Now I'm getting a black screen, flashing it doesn't help. Help me please.

**Comment by Ralim** [comment]
Hello,
Can you please provide a lot more detail; including what files, and tools you used?

You can't hard brick a pinecil; the DFU is burner into the silicon. So this just suggests a bad flash or a flash of the wrong firmware.

**Reply by korzhan** [reply]
I used dfu-util on macOS. I installed the latest IronOS firmware.

**Reply by Ralim** [reply]
Hello, can you please provide more information.
Where did you buy the iron from?
Exactly what command was used?
Exactly which file was used?
What steps you did?
What the output of the dfu until command was?

**Reply by korzhan** [reply]
I bought the soldering iron on the off-site Pine.
I don’t remember exactly what command I used to flash the soldering iron, I just remember that, due to my mistake, it was a bin file and not a dfu file, although this is my first revision.
After that I tried to reflash the dfu file, but nothing helped.
I am attaching the latest logs from the last attempt to flash 2.21 through the PINE64 updater.

```
Downloading firmware Pinecil_EN.dfu
Dowloading completed
dfu-util 0.11-msvc
Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.Copyright 2010-2021 Tormod Volden and Stefan SchmidtThis program is Free Software and has ABSOLUTELY NO WARRANTYPlease report bugs to http://sourceforge.net/p/dfu-util/tickets/
Multiple alternate interfaces for DfuSe file
Opening DFU capable USB device...
Device ID 28e9:0189
Device
 DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Interface #0 ...
Determining device status...
DFU state(2) = dfuIDLE, status(0) = No error condition is present
DFU mode device DFU version 011a
Device returned transfer size 2048
DfuSe interface name: "Internal Flash  "
Found GD32VF103, which reports a bad page size and count for its internal memory.
Fixed layout based on part number: page size 1024, count 128.
File contains 1 DFU images
Parsing DFU image 1
Target name: ST...
Image for alternate setting 0, 
(1 elements, 
total size = 53648)
Setting Alternate Interface #0 ...
Parsing element 1, 
address = 0x08000000, 
size = 53640
Done parsing DfuSe file
```

**Reply by Ralim** [reply]
This is all correct and should work. bin or DFU, both will work. Dfu is mostly useful to prevent you flashing to the wrong thing.

I would suggest trying https://github.com/Spagett1/PineFlash to remove human error on the command as you didn't provide the command used.

**Reply by korzhan** [reply]
I get an error message when flashing the firmware:
```
Pineflash v0.5.3 Windows
PineFlash: Versions successfully fetched.
Pineflash: Pinecil detected
PineFlash: Download of Language Info Complete.
PineFlash: Downloading Firmware Pinecil v2.21.
PineFlash: Download Complete.
PineFlash: File extracted successfully
PineFlash: Attempting to flash Pinecil with the firmware C:\Users\Korzhan\AppData\Local\Temp\v2.21-Pinecil\Pinecil_EN.dfu
Dfu-Util: dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Match vendor ID from file: 28e9
Match product ID from file: 0189
Cannot open DFU device 28e9:0189 found on devnum 6 (LIBUSB_ERROR_NOT_SUPPORTED)
No DFU capable USB device available
```

**Reply by ia** [reply]
Interesting.
> I get an error message when flashing the firmware:

It seems it goes well until this moment:
> Cannot open DFU device 28e9:0189 found on devnum 6 (_**LIBUSB_ERROR_NOT_SUPPORTED**_)

What version of Windows do you have exactly? 10?

TL;DR: basic googling tells me that you may need to install _WinUSB_ driver (or some _libusbK_ or _libusb0_) to make `Pineflash` working.

`Pineflash` using `dfu-util` which using `libusb` - this is a 3rd party library to access USB devices. However, like it mentioned [**here**](https://stackoverflow.com/a/72068953) & [**here**](https://stackoverflow.com/a/17350543), libusb under Windows is OK to get access to connected usb devices and even to get information about them. And these lines from the output is a proof that it works:

> Match vendor ID from file: 28e9
> Match product ID from file: 0189

But it seems `libusb` is not capable in writing/sending data to a usb device under Windows [10] for some reason properly so to do that you need additional drivers.

In fact, it may be a possible minor issue of `Pineflash` version for Windows not mentioning this case in their instructions/HOWTO _if it's the case_.

And [**here is the more detailed instruction**](https://github.com/libusb/libusb/wiki/Windows#user-content-How_to_use_libusb_on_Windows) from `libusb` project of how to install `libusb`-compatible drivers on Windows to make the library working properly under Windows.

P.S. I don't even own `Pinecil` nor have Windows 10 on my hands right now so I may be wrong, just googled error and re-told the results :)

**Reply by korzhan** [reply]
I installed the drivers and now the firmware is loaded successfully into the soldering iron. But it's still a brick and won't start.
```
Pineflash v0.5.3 Windows
PineFlash: Versions successfully fetched.
Pineflash: Pinecil detected
PineFlash: Download of Language Info Complete.
PineFlash: Downloading Firmware Pinecil v2.21.
PineFlash: Download Complete.
PineFlash: File extracted successfully
PineFlash: Attempting to flash Pinecil with the firmware C:\Users\Korzhan\AppData\Local\Temp\v2.21-Pinecil\Pinecil_EN.dfu
Dfu-Util: dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Match vendor ID from file: 28e9
Match product ID from file: 0189
Multiple alternate interfaces for DfuSe file
Opening DFU capable USB device...
Device ID 28e9:0189
Device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Interface #1 ...
Determining device status...
DFU state(2) = dfuIDLE, status(0) = No error condition is present
DFU mode device DFU version 011a
Device returned transfer size 2048
DfuSe interface name: "Option Bytes  "
DfuSe interface name: "Internal Flash  "
Found GD32VF103, which reports a bad page size and count for its internal memory.
Fixed layout based on part number: page size 1024, count 128.
File contains 1 DFU images
Parsing DFU image 1
Target name: ST...
Image for alternate setting 0, (1 elements, total size = 53648)
Setting Alternate Interface #0 ...
Parsing element 1, address = 0x08000000, size = 53640

Erase     [                         ]   0%            0 bytes
Erase     [                         ]   0%            0 bytes
Erase     [=                        ]   7%         4096 bytes
Erase     [==                       ]  11%         6144 bytes
Erase     [===                      ]  15%         8192 bytes
Erase     [====                     ]  19%        10240 bytes
Erase     [=====                    ]  22%        12288 bytes
Erase     [======                   ]  26%        14336 bytes
Erase     [=======                  ]  30%        16384 bytes
Erase     [========                 ]  34%        18432 bytes
Erase     [=========                ]  38%        20480 bytes
Erase     [==========               ]  41%        22528 bytes
Erase     [===========              ]  45%        24576 bytes
Erase     [============             ]  49%        26624 bytes
Erase     [=============            ]  53%        28672 bytes
Erase     [==============           ]  57%        30720 bytes
Erase     [===============          ]  61%        32768 bytes
Erase     [================         ]  64%        34816 bytes
Erase     [=================        ]  68%        36864 bytes
Erase     [==================       ]  72%        38912 bytes
Erase     [===================      ]  76%        40960 bytes
Erase     [====================     ]  80%        43008 bytes
Erase     [=====================    ]  87%        47104 bytes
Erase     [======================   ]  91%        49152 bytes
Erase     [=======================  ]  95%        51200 bytes
Erase     [======================== ]  99%        53248 bytes
Erase     [=========================] 100%        53640 bytes
Erase    done.
Download  [                         ]   0%            0 bytes
Download  [=                        ]   7%         4096 bytes
Download  [==                       ]  11%         6144 bytes
Download  [===                      ]  15%         8192 bytes
Download  [====                     ]  19%        10240 bytes
Download  [=====                    ]  22%        12288 bytes
Download  [======                   ]  26%        14336 bytes
Download  [=======                  ]  30%        16384 bytes
Download  [========                 ]  34%        18432 bytes
Download  [=========                ]  38%        20480 bytes
Download  [==========               ]  41%        22528 bytes
Download  [===========              ]  45%        24576 bytes
Download  [============             ]  49%        26624 bytes
Download  [=============            ]  53%        28672 bytes
Download  [==============           ]  57%        30720 bytes
Download  [===============          ]  61%        32768 bytes
Download  [================         ]  64%        34816 bytes
Download  [=================        ]  68%        36864 bytes
Download  [==================       ]  72%        38912 bytes
Download  [===================      ]  76%        40960 bytes
Download  [====================     ]  80%        43008 bytes
Download  [=====================    ]  87%        47104 bytes
Download  [======================   ]  91%        49152 bytes
Download  [=======================  ]  95%        51200 bytes
Download  [======================== ]  99%        53248 bytes
Download  [=========================] 100%        53640 bytes
Download done.
Done parsing DfuSe file
```

**Reply by ia** [reply]
> But it's still a brick and won't start.

Now it starts to look like [**this issue**](https://github.com/Ralim/IronOS/issues/1661). Let me go a little back in this conversation.

> I bought the soldering iron on the off-site Pine.

When did you buy it exactly? As far as I've heard, the first revision aka PinecilV1 has been discontinued for sale by Pine about year ago. So, just to be clear & to make sure: are you sure that you have PinecilV1 and not PinecilV2? Yours comes with blue rubber holder, not green one, right?

But according to this line in the logs:

> Found **GD32VF103**, which reports a bad page size and count for its internal memory.

it's indeed PinecilV1.

At least your device seems flashable so maybe just like it was suggested for the issue above but [**here**](https://github.com/Ralim/IronOS/issues/1774), could you try to flash release of version 2.20 first? Maybe it will help, maybe it won't. That's what I would do myself anyway - just try to use previous stable release build to see what happen :(

**Reply by korzhan** [reply]
I bought the soldering iron on the off-site https://pine64.com.
The first version with a blue handle.
I bought it November 28, 2021.
I tried installing firmware up to 2.16v. No result.

**Reply by ia** [reply]
Oookay, then it looks like [**another duplicate**](https://github.com/Ralim/IronOS/discussions/988).
Unfortunately, it seems more than couple users had this issue in the past so it may be it. Discussion above looks very interesting.

TL;DR:
- OLED screen should be resoldered (without replacing)
- OLED screen should be replaced

To check the latter, you can unplug your Pinecil after flashing from PC, plug it to reliable power source (just be very careful since tip may be hot), press "+" a few times like you're increasing temperature and put solder on the tip - if solder will be melting, then it's just OLED itself and it should be replaced.

Otherwise, maybe if you disassemble Pinecil and provide photos of soldering joints of OLED cable, more experienced people could tell you if it's the issue (just like in the thread above).

See this replies in particular for more details: [1](https://github.com/Ralim/IronOS/discussions/988#discussioncomment-824591) [2](https://github.com/Ralim/IronOS/discussions/988#discussioncomment-1877480) [3](https://github.com/Ralim/IronOS/discussions/988#discussioncomment-1877548) [4](https://github.com/Ralim/IronOS/discussions/988#discussioncomment-1879345) [5](https://github.com/Ralim/IronOS/discussions/988#discussioncomment-4626857)

Sorry to tell you bad news :(
